packages/core/src/test/shared/utilities/pathFind.test.ts

```diff
@@ -42,4 +41,90 @@ describe('pathFind', function () {
         const regex = /bin[\\\/](code|code-insiders)$/
         assert.ok(regex.test(vscPath), `expected regex ${regex} to match: "${vscPath}"`)
     })
+
+    describe('findSshPath', function () {
+        let previousEnv: NodeJS.ProcessEnv
+
+        beforeEach(function () {
+            previousEnv = testutil.copyEnv()
+        })
+
+        afterEach(function () {
+            process.env = previousEnv
+        })
+
+        it('first tries ssh in $PATH (Non-Windows)', async function () {
+            // skip on windows because ssh in path will never work with .exe extension.
+            if (isWin()) {
+                return
+            }
+            const workspace = await testutil.createTestWorkspaceFolder()
+            const fakeSshPath = path.join(workspace.uri.fsPath, `ssh`)
+
+            process.env = testutil.envWithNewPath(workspace.uri.fsPath)
+            const firstResult = await findSshPath(false)
+
+            await testutil.createExecutableFile(fakeSshPath, '')
+
+            const secondResult = await findSshPath(false)
+
+            assert.notStrictEqual(firstResult, secondResult)
+            assert.strictEqual(secondResult, 'ssh')
+        })
+
+        it('only returns valid executable ssh path (Non-Windows)', async function () {
+            if (isWin()) {
+                return
+            }
+            // On non-windows, we can overwrite path and create our own executable to find.
+            const workspace = await testutil.createTestWorkspaceFolder()
+            const fakeSshPath = path.join(workspace.uri.fsPath, `ssh`)
+
+            process.env = testutil.envWithNewPath(workspace.uri.fsPath)
+
+            await testutil.createExecutableFile(fakeSshPath, '')
+
+            const ssh = await findSshPath(false)
+            assert.ok(ssh)
+            const result = await tryRun(ssh, [], 'yes')
+            assert.ok(result)
+        })
+
+        it('caches result from previous runs (Non-Windows)', async function () {
+            if (isWin()) {
+                return
+            }
+            // On non-windows, we can overwrite path and create our own executable to find.
+            const workspace = await testutil.createTestWorkspaceFolder()
+            // We move the ssh to a temp directory temporarily to test if cache works.
+            const fakeSshPath = path.join(workspace.uri.fsPath, `ssh`)
+
+            process.env = testutil.envWithNewPath(workspace.uri.fsPath)
```

Comment:
since we are now shallow-copying `env` in beforeEach, we can simply directly assign `env.PATH` here. `envWithNewPath` was only needed when beforeEach was not shallow-copied.
```suggestion
            process.env.PATH = workspace.uri.fsPath
```